src/Parser/Mage/Fire/Modules/Features/HotStreak.js

```diff
@@ -0,0 +1,146 @@
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage, formatMilliseconds } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+
+const debug = false;
+
+const PROC_WINDOW_MS = 100;
+
+const HOT_STREAK_CONTRIBUTORS = [
+  SPELLS.FIREBALL.id,
+  SPELLS.PYROBLAST.id,
+  SPELLS.FIRE_BLAST.id,
+  SPELLS.SCORCH.id,
+];
+
+class HotStreak extends Analyzer {
+  static dependencies = {
+		combatants: Combatants,
+	}
+
+  totalProcs = 0;
+  wastedCrits = 0;
+  expiredProcs = 0;
+  pyroWithoutProc = 0;
+  lastCastTimestamp = 0;
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id && spellId !== SPELLS.FLAMESTRIKE.id) {
+      return;
+    }
+    this.lastCastTimestamp = this.owner.currentTimestamp;
+    if (spellId === SPELLS.PHOENIXS_FLAMES.id) {this.phoenixFlamesHits = 0;}
+    if (spellId === SPELLS.DRAGONS_BREATH.id) {this.dragonsBreathHits = 0;}
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if ((!HOT_STREAK_CONTRIBUTORS.includes(spellId) && spellId !== SPELLS.PHOENIXS_FLAMES.id && spellId !== SPELLS.DRAGONS_BREATH.id) || event.hitType !== HIT_TYPES.CRIT || !this.combatants.selected.hasBuff(SPELLS.HOT_STREAK.id)) {
+      return;
+    }
+      //If a direct damage ability crits while the player already has Hot Streak then a proc was overwritten
+      if (HOT_STREAK_CONTRIBUTORS.includes(spellId) && event.hitType === HIT_TYPES.CRIT) {
+        this.wastedCrits += 1;
+        if(debug) { console.log("Hot Streak overwritten @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time)); }
+      }
+      //If the spell was Phoenix Flames, only check the first hit of Phoenix Flames and not the chains/splashes
+      if (spellId === SPELLS.PHOENIXS_FLAMES.id) {
+        if (this.phoenixFlamesHite === 0) {
+          this.wastedCrits += 1;
+          this.phoenixsFlamesHits += 1;
+          if(debug) { console.log("Hot Streak overwritten @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time)); }
+        } else {
+          this.phoenixsFlamesHits += 1;
+        }
+      }
+      //If the spell was Dragon's Breath, only count the first hit and only if the player has Alexstrasza's Fury.
+      if (spellId === SPELLS.DRAGONS_BREATH.id && this.combatants.selected.hasTalent(SPELLS.ALEXSTRASZAS_FURY_TALENT.id)) {
+        if (this.dragonsBreathHits === 0) {
+          this.wastedCrits += 1;
+          this.dragonsBreathHits += 1;
+          if(debug) { console.log("Hot Streak overwritten @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time)); }
+        } else {
+          this.dragonsBreathHits += 1;
+        }
+      }
```

Comment:
``dragonsBreathHits`` should have a ``dragonsBreathHits = 0;`` at the top aswell